mapie/multi_label_classification.py

```diff
@@ -0,0 +1,746 @@
+        time the partial_fit method is called.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Training data.
+
+        y : NDArray of shape (n_samples, n_classes)
+            Training labels.
+
+        sample_weight : Optional[ArrayLike] of shape (n_samples,)
+            Sample weights for fitting the out-of-fold models.
+            If None, then samples are equally weighted.
+            If some weights are null,
+            their corresponding observations are removed
+            before the fitting process and hence have no prediction sets.
+
+            By default None.
+
+        Returns
+        -------
+        MapieMultiLabelClassifier
+            The model itself.
+        """
+        # Checks
+        first_call = self._check_partial_fit_first_call()
+        self._check_parameters()
+        estimator = self._check_estimator(X, y, self.estimator)
+
+        X, y = indexable(X, y)
+        _check_y(y, multi_output=True)
+        sample_weight, X, y = check_null_weight(sample_weight, X, y)
+        y = cast(NDArray, y)
+        X = cast(NDArray, X)
+
+        self._check_all_labelled(y)
+        self.n_samples_ = _num_samples(X)
+
+        # Work
+        self.single_estimator_ = estimator
+        y_pred_proba = self.single_estimator_.predict_proba(X)
+        y_pred_proba_array = self._transform_pred_proba(y_pred_proba)
+        if first_call:
+            self.theta_ = np.zeros(X.shape[1])
+            self.risks = self._compute_risks(y_pred_proba_array, y)
+        else:
+            if X.shape[1] != self.theta_.shape[0]:
+                msg = "Number of features %d does not match previous data %d."
+                raise ValueError(msg % (X.shape[1], self.theta_.shape[0]))
+            partial_risk = self._compute_risks(y_pred_proba_array, y)
+            self.risks = np.concatenate([self.risks, partial_risk], axis=0)
+
+        return self
+
+    def fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        sample_weight: Optional[ArrayLike] = None,
+    ) -> MapieMultiLabelClassifier:
+        """
+        Fit the base estimator or use the fitted base estimator.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Training data.
+
+        y : NDArray of shape (n_samples, n_classes)
+            Training labels.
+
+        sample_weight : Optional[ArrayLike] of shape (n_samples,)
+            Sample weights for fitting the out-of-fold models.
+            If None, then samples are equally weighted.
+            If some weights are null,
+            their corresponding observations are removed
+            before the fitting process and hence have no prediction sets.
+
+            By default None.
+
+        Returns
+        -------
+        MapieMultiLabelClassifier
+            The model itself.
+        """
+        if not self._check_partial_fit_first_call():
```

Comment:
should you raise an error instead, and force to create a new instance?

Reply:
replaced by argument _refit, set to true if calling fit according to sklearn procedure (cf https://github.com/scikit-learn/scikit-learn/blob/f3f51f9b611bf873bd5836748647221480071a87/sklearn/naive_bayes.py#L367)